### ISSUE TITLE
SC2016 false positive

### ISSUE THREAD

**Issue by MohamadKh75** [koalaman/shellcheck#2445]
#### For bugs

- Rule Id: SC2016
- My shellcheck version: 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions

- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

find "screenshots/$NATIONAL_CODE" -type f \
  -name "*.jpeg" \
  ! -name "captcha.jpeg" \
  ! -name "processed.jpeg" \
  -print0 |
  sort -z |
  xargs -0 bash -c \
    'convert "$@" pdf/"$NATIONAL_CODE".pdf' _

```

#### Here's what shellcheck currently says:

SC2016 (info): Expressions don't expand in single quotes, use double quotes for that.

#### Here's what I wanted or expected to see:

No warning as it works correctly!

**Comment by kolyshkin** [comment]
Interestingly, shellcheck have some code to ignore the `sh -c` argument, and also does that for docker/podman exec. I guess that `xargs -0` is what prevents it from using it.

**Comment by MohamadKh75** [comment]
I hope we see the fix soon 😁
Thanks for this awesome tool!
